### PR TITLE
Legend state

### LINKED ISSUE #10481: Editing the style of a layer reset's its legend's titles [DEPENDS ON #10904]
**STR**

0. Have map with legends
1. Set a custom title for a legend
2. Change styles of that layer (color or whatever)

**Expected result:**

Styles change and maybe the legend has to change as well, but the title stays the same.

**Actual result:**

The legend is reset and the title is as well.

I guess it's debatable but it has given me some frustration today. What do you guys think? Bug or feature?

cc/ @xavijam

### PR DESCRIPTION
**Pull request by nobuti**
This PR fixes #10481.

legends-custom-state

Still pending the migrations in order to persist the state, so WIP for now.

**Not merge until https://github.com/CartoDB/cartodb/pull/10904**

CR @xavijam

### CODE REVIEW

**Comment by nobuti** [comment]
Back integrated smoothly, ready for review.

**Comment by nobuti** [comment]
Because this depends on #10904 not sure how to put it on staging.

**Comment by xavijam** [comment]
@gfiorav could we "unblock" https://github.com/CartoDB/cartodb/pull/10904 on monday?

**Comment by gfiorav** [comment]
@xavijam asked for CR again. We have to deploy the migration separately. Doable on Monday. 

@nobuti just merge that branch (10877-add-conf-column-to-legends) into yours and upload to staging

**Comment by nobuti** [comment]
In order to keep things separated I'm deploying to ded06 an intermediate branch from this with #10877

**Comment by gfiorav** [comment]
@nobuti I wouldn't worry too much, once the backend part hits prod, the diff won't show those changes.

**Comment by nobuti** [comment]
Ready for review again. You should check:
- [ ] Title remains even the legend type changes.
- [ ] Check that everything works as expected with legends in two (or more) different layers.
- [ ] Check everything while adding / removing layers.

File: lib/assets/javascripts/cartodb3/data/legends/legend-base-definition-model.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 var Backbone = require('backbone');
 var _ = require('underscore');
 var syncAbort = require('../backbone/sync-abort');
+var LegendsSate = require('./legends-state');
```
Comment (javitonino) [inline lib/assets/javascripts/cartodb3/data/legends/legend-base-definition-model.js]: Typo (Sate -> State)

Not much else I can say about this PR, someone else should take a look.

Comment (nobuti) [inline lib/assets/javascripts/cartodb3/data/legends/legend-base-definition-model.js]: Oh, good catch :shipit:

**Pull request merged**